common/src/main/java/org/opensearch/ml/common/parameter/MLModel.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.parameter;
+
+import lombok.Builder;
+import lombok.Getter;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.commons.authuser.User;
+
+import java.io.IOException;
+import java.util.Base64;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+@Getter
+public class MLModel implements ToXContentObject {
+    public static final String ALGORITHM = "algorithm";
+    public static final String MODEL_NAME = "name";
+    public static final String MODEL_VERSION = "version";
+    public static final String MODEL_CONTENT = "content";
+    public static final String USER = "user";
+
+    private String name;
+    private FunctionName algorithm;
+    private Integer version;
+    private String content;
+    private User user;
+
+    @Builder
+    public MLModel(String name, FunctionName algorithm, Integer version, String content, User user) {
+        this.name = name;
+        this.algorithm = algorithm;
+        this.version = version;
+        this.content = content;
+        this.user = user;
+    }
+
+    public MLModel(FunctionName algorithm, Model model) {
+        this(model.getName(), algorithm, model.getVersion(), Base64.getEncoder().encodeToString(model.getContent()), null);
+    }
+
+    public MLModel(StreamInput input) throws IOException{
+        name = input.readOptionalString();
```

Comment:
Keep the same order when read from `StreamInput` as write out `StreamOutput`. The `writeTo` method write `content` after `name`, so we should read `content` after `name`.

Reply:
Yeah will do in the revision.